compiler/rustc_expand/src/derive_macro_expansion.rs

```diff
@@ -0,0 +1,125 @@
+use std::cell::Cell;
```

Comment:
It's probably better to merge this file into `compiler/rustc_expand/src/proc_macro.rs` and put the new content in the bottom.

Reply:
Done; I think `with_context` and `enter_context` could maybe use more descriptive names soon, since they are now in a bigger file. But I'd wait until attr- and bang-proc-macros are also cacheable, to see if it's reused for those.